Source/ResponseSerialization.swift

```diff
@@ -64,18 +64,42 @@ public protocol DownloadResponseSerializerProtocol {
 
 /// A serializer that can handle both data and download responses.
 public protocol ResponseSerializer: DataResponseSerializerProtocol & DownloadResponseSerializerProtocol {
+    /// `DataPreprocessor` closure used to prepare incoming `Data` for serialization.
+    var dataPreprocessor: DataPreprocessor { get }
     /// `HTTPMethod`s for which empty response bodies are considered appropriate.
     var emptyRequestMethods: Set<HTTPMethod> { get }
     /// HTTP response codes for which empty response bodies are considered appropriate.
     var emptyResponseCodes: Set<Int> { get }
 }
 
+/// Type used to preprocess `Data` before it handled by a serializer.
+public protocol DataPreprocessor {
+    /// Process           `Data` before it's handled by a serializer.
+    /// - Parameter data: The raw `Data` to process.
+    func preprocess(_ data: Data) throws -> Data
+}
+
+/// `DataPreprocessor` that returns passed `Data` without any transform.
+struct Passthrough: DataPreprocessor {
+    func preprocess(_ data: Data) throws -> Data { return data }
+}
+
+/// `DataPreprocessor` that trims Google's typical `)]}',\n` XSSI JSON header.
+struct XSSI: DataPreprocessor {
+    func preprocess(_ data: Data) throws -> Data {
+        return (data.prefix(6) == Data(")]}',\n".utf8)) ? data.dropFirst(6) : data
+    }
+}
```

Comment:
I'm not sure this `XSSI` type is actually being used at all right now. Also, would it make sense to expose `Passthrough` as a `public` type? Would it be convenient for clients, or would you just expect those building custom response serializers to use `ResponseSerializer.defaultDataPreprocessor`?

Reply:
If either of these do get made `public`, I think I'd recommend tacking on a suffix like `Preprocessor` or `DataPreprocessor` so it's a bit more clear what they are.

Reply:
I've made them both public, updated the naming, and added some tests.